app/src/main/java/app/revanced/manager/ui/component/bundle/BundlePatchesDialog.kt

```diff
@@ -62,44 +63,213 @@ fun BundlePatchesDialog(
                         )
                     }
                 }
+            }
+        }
+    }
+}
 
-                state.patchBundleOrNull()?.let { bundle ->
-                    items(bundle.patches.size) { bundleIndex ->
-                        val patch = bundle.patches[bundleIndex]
-                        ListItem(
-                            headlineContent = {
-                                Text(
-                                    text = patch.name,
-                                    style = MaterialTheme.typography.bodyLarge,
-                                    color = MaterialTheme.colorScheme.onSurface
-                                )
-                            },
-                            supportingContent = {
-                                patch.description?.let {
-                                    Text(
-                                        text = it,
-                                        style = MaterialTheme.typography.bodyMedium,
-                                        color = MaterialTheme.colorScheme.onSurfaceVariant
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun PatchItem(
+    patch: PatchInfo,
+    expandVersions: Boolean,
+    onExpandVersions: () -> Unit,
+    expandOptions: Boolean,
+    onExpandOptions: () -> Unit
+) {
+    ElevatedCard(
+        modifier = Modifier
+            .fillMaxWidth()
+            .then(
+                if (patch.options.isNullOrEmpty()) Modifier else Modifier
+                    .clip(RoundedCornerShape(8.dp))
+                    .clickable(onClick = onExpandOptions),
+            )
+    ) {
+        Column(
+            modifier = Modifier.padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(6.dp)
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Absolute.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = patch.name,
+                    color = MaterialTheme.colorScheme.primary,
+                    style = MaterialTheme.typography.titleLarge,
+                    fontWeight = FontWeight.Bold
+                )
+
+                if (!patch.options.isNullOrEmpty()) {
+                    ArrowButton(expanded = expandOptions)
+                }
+            }
+            patch.description?.let {
+                Text(
+                    text = it,
+                    style = MaterialTheme.typography.bodyMedium
+                )
+            }
+            Column(
+                verticalArrangement = Arrangement.spacedBy(4.dp)
+            ) {
+                if (!patch.compatiblePackages.isNullOrEmpty()) {
+                    patch.compatiblePackages.forEach { compatiblePackage ->
+                        val packageName = compatiblePackage.packageName
+                        val versions = compatiblePackage.versions.orEmpty().reversed()
```

Comment:
Why are the versions reversed?

Reply:
Because when patch devs add new versions, they add them to the end of the list rather than the beginning